Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/contest-solution-submit/ContestSolutionSubmitPage.tsx

```diff
@@ -0,0 +1,474 @@
+                    <div>
+                        <span>Allowed extensions:</span>
+                        {' '}
+                        {allowedFileExtensions.join(', ')}
+                    </div>
+                    {fileUploadError && <div className={styles.fileUploadError}>{fileUploadError}</div>}
+                    <FileUploader
+                      file={uploadedFile}
+                      problemId={id}
+                      allowedFileExtensions={allowedFileExtensions}
+                      onInvalidFileExtension={(e) => setFileUploadError(e.detail)}
+                      onFileUpload={(file) => {
+                          setFileUploadError('');
+                          setUploadedFile(file);
+                      }}
+                    />
+                    <Button
+                      className={styles.fileSubmitButton}
+                      onClick={onSolutionSubmitFile}
+                      text="Submit"
+                      state={!uploadedFile || fileUploadError || !submitButtonIsAvailable
+                          ? ButtonState.disabled
+                          : ButtonState.enabled}
+                    />
+                </div>
+            );
+        }
+
+        return (
+            <div>
+                <CodeEditor
+                  selectedSubmissionType={selectedSubmissionType}
+                  code={submissionCode}
+                  onCodeChange={(inputCode) => setSubmissionCode(inputCode)}
+                />
+                <div className={styles.submitSettings}>
+                    <Dropdown
+                      dropdownItems={strategyDropdownItems}
+                      value={selectedStrategyValue}
+                      handleDropdownItemClick={onStrategyDropdownItemSelect}
+                    />
+                    <Button
+                      state={!submitButtonIsAvailable
+                          ? ButtonState.disabled
+                          : ButtonState.enabled}
+                      onClick={onSolutionSubmitCode}
+                      text="Submit"
+                    />
+                </div>
+            </div>
+        );
+    };
+
+    if (isLoading || userParticipationDataLoading) {
+        return <div style={{ ...flexCenterObjectStyles }}><SpinningLoader /></div>;
+    }
+
+    if (error || userParticipationError) {
+        return (
+            <div className={styles.contestSolutionSubmitWrapper}>
+                <div className={textColorClassName}>Error fetching user participation data!</div>
+            </div>
+        );
+    }
+
+    if (requirePassword && shouldEnterPassword) {
+        return (
+            <ContestPasswordForm
+              id={Number(contestId)}
+              isOfficial={isCompete}
+              contestName={name!}
+              onSuccess={fetchUserParticipationDetails}
+            />
+        );
+    }
+
+    if (isCompete && isOnlineExam && !hasAcceptedOnlineExamModal) {
```

Comment:
Didn't we agree to not rely on isOnline exam logic, instead the back end will return `shouldShowConfirmationModal`

Reply:
forgot about that one, will rework

Reply:
done